### PR TITLE
Fix/toggle login

### PR DESCRIPTION
**Pull request by Aborysa**
#18 
- Made it possible to disable ToggleSwitch (there is no visual queue for this atm thought)
- Fixed state's `checked` property not updating after props have changed.
- Made login toggle reflect the Login component's state properly.

### CODE REVIEW

File: app/src/components/ToggleSwitch.jsx [hunk 1]
```diff
@@ -10,21 +10,33 @@ class ToggleSwitch extends React.Component {
...
+  componentWillReceiveProps(props){
+    this.setState({
+      checked: props.checked != null ? props.checked : this.state.checked,
+    });
+  }
+
   handleToggle(e, a) {
     this.setState({
-      checked: !this.state.checked
+      checked: !this.state.checked,
```
Comment (duvholt) [inline app/src/components/ToggleSwitch.jsx]: I don't think this component needs a state anymore. But if it works we can change it later.

File: app/src/components/__tests__/ToggleSwitch.js [hunk 2]
```diff
@@ -19,10 +28,54 @@ describe('ToggleSwitch.jsx', () => {
   });
 
   it('toggles checked when clicked', () => {
-    const wrapper = shallow(
+    const wrapper = mount(
```
Comment (Aborysa) [inline app/src/components/__tests__/ToggleSwitch.js]: Needs mount in order to do test on state.

File: app/src/components/ToggleSwitch.jsx (hunk 1 above)
Comment (Aborysa) [inline app/src/components/ToggleSwitch.jsx]: the "checked" prop is not always provided by the parent component, so for it to work in those cases state is required.

File: app/src/components/__tests__/ToggleSwitch.js (hunk 2 above)
Comment (duvholt) [inline app/src/components/__tests__/ToggleSwitch.js]: Ah, makes sense.

File: app/src/components/ToggleSwitch.jsx (hunk 1 above)
Comment (duvholt) [inline app/src/components/ToggleSwitch.jsx]: Wouldn't defaultProps solve that?

**Pull request merged**